File: flood-it/Flood-it/Flood_it/src/Model/Observable.cpp
#include "Observable.h"

void Observable::addObserver(Observer* observer) {
    observers.push_back(observer);
}

void Observable::removeObserver(Observer* observer) {
    observers.erase(std::remove(observers.begin(), observers.end(), observer), observers.end());
}

void Observable::notifyObservers() {
    for (Observer* observer : observers) {
        observer->update();
    }
}

File: othello/Othello/src/main/java/Observer/Observable.java
package Observer;

import java.util.ArrayList;

/**
 * Observable class is used to notify the observers when a change is made.
 */
public class Observable {
    private ArrayList<Observer> observer;

    public Observable() {
        observer = new ArrayList<>();
    }

    /**
     * Add observer to the list of observers
     *
     * @param observer the observer to be added
     */
    public void addObserver(Observer observer) {
        this.observer.add(observer);
    }

    /**
     * Remove observer from the list of observers
     *
     * @param obj the observer to be removed
     */
    public void notifyObserver(Object obj) {
        for (Observer o : observer) {
            o.update(obj);
        }
    }


}
